### PR TITLE
[FIX] account: add method to load CoA for a specific company to make …

### PR DESCRIPTION
**Pull request by jco-odoo**
…it easier to use for demo data companies

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jco-odoo** [comment]
@zaoral @jjscarafia I have not tested yet, but I think it would make it a lot easier for demo data companies to use this function try_loading_for_company.

**Comment by tivisse** [comment]
At the first sight, it looks good to me.

@jco-odoo Don't forget to test the corner cases before merge ;)

Thanks.

**Comment by zaoral** [comment]
Hi @jco-odoo 

I have tested and now is working. But, we also needs this change in order to properly install in a enterprise environment https://github.com/odoo/enterprise/pull/4975

Best Regards

**Comment by zaoral** [comment]
@jco-odo  please review this PR https://github.com/odoo-dev/odoo/pull/138 as a fix to the current PR

**Comment by jco-odoo** [comment]
@tivisse @zaoral PR updated.   @csnauwaert is the structure ok for you?

**Review by zaoral — APPROVED** [review]
LGTM +1

NOTE: Don't forget to create the related PR in enterprise with the modification of load_for_current_company works (Can re open this PR that alreadu have the changes https://github.com/odoo/enterprise/pull/4975 )

File: addons/account/models/chart_template.py [hunk 1]
```diff
@@ -176,7 +176,15 @@ def try_loading_for_current_company(self):
...
-    def load_for_current_company(self, sale_tax_rate, purchase_tax_rate):
+    def try_loading_for_company(self, company):
+        if not company:
+            self.try_loading_for_current_company()
+        else:
+            if not company.chart_template_id and not self.existing_accounting(company):
+                for template in self:
+                    template.with_context(default_company_id=company.id).load_for_current_company(15.0, 15.0, company)
+
+    def load_for_current_company(self, sale_tax_rate, purchase_tax_rate, company=False):
```
Comment (csnauwaert) [inline addons/account/models/chart_template.py]: I find it strange and confusing that the method is called `load_for_current_company` but can now be called for another company than the current one if you pass a company parameter.
So since we are refactoring why not only keep the 2 following methods: `load_for_company` and `try_loading_for_company` but it means that you'll have to change the code of all chart_templates that were previously calling `try_loading_for_current_company` and `load_for_current_company`. Also the new methods that are taking a company as parameter should works as follow: if no company is provided, assume that we are calling with the current company.
This would be the best solution as we would simplify the code and won't have duplicated method but if you don't want to change the code on each localization, I would be fine keeping the `try_loading_for_current_company` and `load_for_current_company` but those function would just call their counterpart `try_loading_for_company` and `load_for_company` without passing the company parameter.

TLDR; the logic should be inside `try_loading_for_company` and `load_for_company`

**Review by csnauwaert — CHANGES_REQUESTED** [review]

**Comment by jco-odoo** [comment]
Linked PR https://github.com/odoo/enterprise/pull/5071
